New MRS v2 job resource

Lance52259 opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:
Compared with the job of V1 version, the parameter design of V2 version refers to the console, which makes it more convenient for users to use job resources against the console and integrate old parameters such as jar path into a program parameters dictionary.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
reference: #1324 

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->

```release-note
1. New job resource support:
  Seven kinds job: Flink, MapReduce, SparkSubmit, HiveSql, HiveScript, SparkSql and SparkScript.
2. Related document and acc test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/mrs' TESTARGS='-run=TestAccMrsMapReduceJob_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/mrs -v -run=TestAccMrsMapReduceJob_basic -timeout 360m -parallel 4
=== RUN   TestAccMrsMapReduceJob_basic
=== PAUSE TestAccMrsMapReduceJob_basic
=== CONT  TestAccMrsMapReduceJob_basic
--- PASS: TestAccMrsMapReduceJob_basic (70.14s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs       70.215s
```

Lance52259 committed:
feat: new mrs v2 job resource supported

hmbig2 commented on huaweicloud/services/mrs/resource_huaweicloud_mapreduce_job.go:
```diff
@@ -0,0 +1,525 @@
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Flink", "HiveSql", "HiveScript", "MapReduce", "SparkSubmit", "SparkSql", "SparkScript",
```
defined as a constant, it is more convenient to use, such as in test : resource.TestCheckResourceAttr(resourceName, "type", "SparkSubmit"),

hmbig2 commented on docs/resources/mapreduce_job.md:
```diff
@@ -0,0 +1,112 @@
+
+  -> **NOTE:** Spark and Hive jobs can be added to only clusters including Spark and Hive components.
+
+* `jar_path` - (Optional, String, ForceNew) Specifies the .jar package path for program execution.
```
when type =sparkSubmit
 in the console webpage  : this field label is "Program Path",  support .jar or .py

Lance52259 committed:
udpate mrs job resource and documentation

- use constant to defind the job type
- adjust document description
- use program path to replace the jar path parameter
